internal/core/src/query/CachedSearchIterator.h

```diff
@@ -0,0 +1,182 @@
+                         const BitsetView& bitset,
+                         const milvus::DataType& data_type);
+
+    // For growing segment with chunked data, BF
+    CachedSearchIterator(const dataset::SearchDataset& dataset,
+                         const segcore::VectorBase* vec_data,
+                         const int64_t row_count,
+                         const SearchInfo& search_info,
+                         const std::map<std::string, std::string>& index_info,
+                         const BitsetView& bitset,
+                         const milvus::DataType& data_type);
+
+    // For sealed segment with chunked data, BF
+    CachedSearchIterator(const std::shared_ptr<ChunkedColumnBase>& column,
+                         const dataset::SearchDataset& dataset,
+                         const SearchInfo& search_info,
+                         const std::map<std::string, std::string>& index_info,
+                         const BitsetView& bitset,
+                         const milvus::DataType& data_type);
+
+    // This method fetches the next batch of search results based on the provided search information
+    // and updates the search_result object with the new batch of results.
+    void
+    NextBatch(const SearchInfo& search_info, SearchResult& search_result);
+
+    // Disable copy and move
+    CachedSearchIterator(const CachedSearchIterator&) = delete;
+    CachedSearchIterator&
+    operator=(const CachedSearchIterator&) = delete;
+    CachedSearchIterator(CachedSearchIterator&&) = delete;
+    CachedSearchIterator&
+    operator=(CachedSearchIterator&&) = delete;
+
+ private:
+    using DisIdPair = std::pair<float, int64_t>;
+    using IterIdx = size_t;
+    using IterIdDisIdPair = std::pair<IterIdx, DisIdPair>;
+    using GetChunkDataFunc =
+        std::function<std::pair<const void*, int64_t>(int64_t)>;
+
+    int64_t batch_size_ = 0;
+    std::vector<knowhere::IndexNode::IteratorPtr> iterators_;
+    int8_t sign_ = 1;
+    size_t num_chunks_ = 1;
+    size_t nq_ = 0;
+
+    struct IterIdDisIdPairComparator {
+        bool
+        operator()(const IterIdDisIdPair& lhs, const IterIdDisIdPair& rhs) {
+            if (lhs.second.first == rhs.second.first) {
+                return lhs.second.second > rhs.second.second;
+            }
+            return lhs.second.first > rhs.second.first;
+        }
+    };
+    std::vector<std::priority_queue<IterIdDisIdPair,
+                                    std::vector<IterIdDisIdPair>,
+                                    IterIdDisIdPairComparator>>
+        chunked_heaps_;
+
+    inline bool
+    IsValid(const DisIdPair& result,
+            const std::optional<float>& last_bound,
+            const std::optional<float>& radius,
+            const std::optional<float>& range_filter) {
+        const float dist = result.first;
+        const bool is_valid =
+            !last_bound.has_value() || dist > last_bound.value();
+
```

Comment:
question: no need to consider the positive or negative metrics for dist and last_bound?

Reply:
The distances are converted when entering this class, no need to worry about it here